examples/deepzoom/deepzoom_server.py

```diff
@@ -144,6 +150,50 @@ def slugify(text):
     text = normalize('NFKD', text.lower()).encode('ascii', 'ignore').decode()
     return re.sub('[^a-z0-9]+', '-', text)
 
+def icc_apply(img, fname):
+    icc_profile = get_icc_profile(fname)
+    if icc_profile is None:
+        return img
+    else:
+        return ImageCms.applyTransform(img, icc_profile)
+    
+    
+@lru_cache()
+def get_icc_profile(fname):
+    #Source: http://www.andrewjanowczyk.com/application-of-icc-profiles-to-digital-pathology-images/
+    
+    if fname is None:
+        fname = app.config['DEEPZOOM_SLIDE']
+
+    ImageFile.LOAD_TRUNCATED_IMAGES = True
+ 
+    #Need to set this to none, otherwise PIL raises an error as its concerned our image is too big and is in fact a decompression bomb
+    Image.MAX_IMAGE_PIXELS = None
+
+    try:
+        icc = Image.open(fname).info.get('icc_profile') 
```

Comment:
You can't open an arbitrary slide with Pillow and expect it to work.  Some slide files are TIFF, some are TIFF-like but can't be read by standard TIFF readers, and some are in proprietary formats that Pillow can't read.  That's why OpenSlide exists in the first place.

https://github.com/openslide/openslide/issues/139 covers adding an OpenSlide API method to read the slide's ICC profile.  Recently there's been some interest in implementing it, and it may happen soon.